### ISSUE TITLE
No transitions on first slide while using navigation buttons when destroy and reinit the carousel

### ISSUE THREAD

**Issue by sadeghbarati** [davidjerleke/embla-carousel#672]
### Bug is related to

- [x] embla-carousel (core package)
- [ ] embla-carousel-react
- [x] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs (documentation)
- [ ] embla-carousel-docs (generator)

### Embla Carousel version

- v8.0.0-rc17

### Describe the bug

No transition occurs on first slide on hitting `next` button when destroy and reinit the Embla Carousel

### CodeSandbox / StackBlitz

https://stackblitz.com/edit/vitejs-vite-k4menh?file=src%2Fcomponents%2FCarousel.vue

### Steps to reproduce

1. Go to stackblitz repro
2. Click on `Destroy/Reinit Carousel`
3. Click `next` button
4. No transitions occurs on first slide

### Expected behavior

slide should have normal transitions

### Additional context

None

**Comment by sadeghbarati** [comment]
https://github.com/shadcn-ui/ui/issues/2230

**Comment by sadeghbarati** [comment]
It's working fine now 🙏

What was the problem

**Comment by davidjerleke** [comment]
@sadeghbarati you'll see when I create the PR. I will release a patch release with the bug fix as soon as possible. It will be `v8.0.0-rc18`. Thanks for testing it!

**Comment by sadeghbarati** [comment]
Just one thing before the publish Vue


Isn't better to use `onBeforeUnmount` instead of `onUnmounted`?

**Comment by davidjerleke** [comment]
@sadeghbarati yeah maybe that’s better? I’m not a seasoned Vue dev. Most of my experience is with React.

**Comment by sadeghbarati** [comment]
Both are working well just to separate the embla destroying process and Vue reactivity cleanup

**Comment by davidjerleke** [comment]
@sadeghbarati feel free to try it out in the latest [StackBlitz fork](https://github.com/davidjerleke/embla-carousel/issues/672#issuecomment-1872933088) I shared and submit a PR with the onBeforeUnmount hook instead of onUnmounted.

Best,
David

**Comment by sadeghbarati** [comment]
Thank You ❤️‍🔥

**Comment by davidjerleke** [comment]
@sadeghbarati so you will create a PR 🙂?

**Comment by sadeghbarati** [comment]
Sure thing! 

But It was easy to contribute if I was pnpm and pnpm workspaces :grin:

**Comment by sadeghbarati** [comment]
Sry it took too long I was trying to figure out the yarn and creating playground for Vue

Should I have to include `embla-carousel-playground-vue` in this PR https://github.com/davidjerleke/embla-carousel/pull/673?

**Comment by davidjerleke** [comment]
@sadeghbarati thanks for the PR 👍🏻! The build files differ [because of this upcoming change](https://github.com/davidjerleke/embla-carousel/issues/647#issuecomment-1868273239) which I haven’t released yet.

I think the playground should be a separate PR just to separate things. You can check out the vanilla and react playgrounds to see how the styles are imported from the documentation website examples. You don’t need to add your own that way. And I would recommend you to setup a Vite playground just like the vanilla and react to make it easier for devs to jump between the playgrounds when testing things.

Thanks for your efforts!

**Comment by sadeghbarati** [comment]
Ok will create new PR for playground later after the new release

Thanks for the opportunity ❤️

**Comment by davidjerleke** [comment]
@sadeghbarati a bug fix for this was just released in [v8.0.0-rc18](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc18). Try it out and let me know if it's working as expected.
